Controller/Payment/CancelInPagePayment.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Alma\MonthlyPayments\Controller\Payment;
+
+use Alma\API\RequestError;
+use Alma\MonthlyPayments\Gateway\Config\Config;
+use Alma\MonthlyPayments\Helpers\AlmaClient;
+use Alma\MonthlyPayments\Helpers\Exceptions\AlmaClientException;
+use Alma\MonthlyPayments\Helpers\Logger;
+use Alma\MonthlyPayments\Helpers\OrderHelper;
+use Magento\Checkout\Model\Session;
+use Magento\Framework\App\ActionInterface;
+use Magento\Framework\Controller\Result\Json;
+use Magento\Framework\Controller\Result\JsonFactory;
+use Magento\Sales\Model\Order;
+
+class CancelInPagePayment implements ActionInterface
+{
+    /**
+     * @var Logger
+     */
+    private $logger;
+    /**
+     * @var JsonFactory
+     */
+    private $jsonFactory;
+    /**
+     * @var Session
+     */
+    private $session;
+    /**
+     * @var OrderHelper
+     */
+    private $orderHelper;
+    private AlmaClient $almaClient;
+
+    /**
+     * @param Logger $logger
+     * @param JsonFactory $jsonFactory
+     * @param Session $session
+     * @param OrderHelper $orderHelper
+     */
+    public function __construct(
+        Logger $logger,
+        JsonFactory $jsonFactory,
+        Session $session,
+        OrderHelper $orderHelper,
+        AlmaClient $almaClient
+    ) {
+        $this->logger = $logger;
+        $this->jsonFactory = $jsonFactory;
+        $this->session = $session;
+        $this->orderHelper = $orderHelper;
+        $this->almaClient = $almaClient;
+    }
+
+    /**
+     * @return Json
+     */
+    public function execute(): Json
+    {
+        $response = $this->jsonFactory->create();
+        $order = $this->session->getLastRealOrder();
+        if (!$order) {
+            $this->error($response, 'No Order in session');
+            return $response;
+        }
+        if (!$order->canCancel()) {
+            $this->error($response, 'Order can not be cancelled');
+            return $response;
+        }
+        $order->cancel()->addStatusToHistory(Order::STATE_CANCELED, 'In Page Payment canceled by customer');
```

Comment:
typo error on State_Canceled

Reply:
it's a default magento State 
    public const STATE_CANCELED = 'canceled';